pkg/config/metadata/user_info_test.go

```diff
@@ -0,0 +1,167 @@
+package metadata
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"gopkg.in/yaml.v2"
+	"gotest.tools/assert"
+
+	"github.com/3scale-labs/authorino/pkg/common"
+	"github.com/3scale-labs/authorino/pkg/config/identity"
+
+	envoy_auth "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"
+)
+
+const (
+	authServerHost string = "127.0.0.1:9001"
+	userInfoClaims string = `{ "sub": "831707be-ef07-4d63-b427-4216309e9897" }`
+)
+
+var (
+	rawAPIConfig string = fmt.Sprintf(`
+identity:
+  - name: auth-server
+    oidc:
+      endpoint: http://%s
+metadata:
+  - name: userinfo-auth-server
+    userinfo:
+      oidc: auth-server
+      client_id: my_client
+      client_secret: do_not_tell_anyone
+`, authServerHost)
+
+	wellKnownOIDCConfig string = fmt.Sprintf(`{
+		"issuer": "http://%s",
+		"userinfo_endpoint": "http://%s/userinfo"
+	}`, authServerHost, authServerHost)
+
+	authContext *__AuthContext
```

Comment:
Maybe in the near future we could introduce factories for mock objects... specially with the `AuthContext` one (which we might want to choose a better name then)